core/ir/aarch64/decode.c

```diff
@@ -89,6 +91,15 @@ byte *
 decode(void *drcontext, byte *pc, instr_t *instr)
 {
     dcontext_t *dcontext = (dcontext_t *)drcontext;
+
+    /* Synthetic ISA has its own decoder.
+     * XXX i#1684: when DR can be built with full dynamic architecture selection we won't
+     * need to pollute the decoding of other architectures with this synthetic ISA special
+     * case.
+     */
+    if (dr_get_isa_mode(drcontext) == DR_ISA_SYNTHETIC)
```

Comment:
Shouldn't this live in decode_common?  Many users (including probably every user decoding from a memtrace) will use decode_from_copy and completely bypass this.

Reply:
Done.
Moved to `decode_common()`.